doc/topics/blocking-vs-non-blocking.md

```diff
@@ -0,0 +1,143 @@
+# Overview of Blocking vs Non-Blocking
+
+This overview covers the difference between **blocking** and **non-blocking**
```

Comment:
@tomgco I put this note at the top of the `Comparing Code` section prior to showing the code samples